### PR TITLE
[21136] Add XML configuration for FlowControllerDescriptor to 2.x

### PR DESCRIPTION
**Pull request by Mario-DL**
## Description

This PR is a partial backport from #4837 that brings the XML configuration of the FLow Controller Descriptor List. In `2.x` both `ThroughtputController` and `FlowcontrollerDescriptor` coexist.

@Mergifyio backport 2.13.x 2.10.x 2.6.x

## Contributor Checklist

- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
- [X] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [X] Any new/modified methods have been properly documented using Doxygen. 
- [X] Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- [ X Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- [X] Changes are API compatible. 
- [X] New feature has been added to the `versions.md` file (if applicable).
- [X] New feature has been documented/Current behavior is correctly described in the documentation. 
    - Related documentation PR: eProsima/Fast-DDS-docs#793
- [X] Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: test/unittest/dds/profiles/test_xml_profile.xml [hunk 1]
```diff
@@ -288,6 +300,18 @@
                     <affinity>15</affinity>
                     <stack_size>1048576</stack_size>
                 </security_log_thread>
+                <flow_controller_descriptor_list>
+                    <flow_controller_descriptor>
+                        <name>test_flow_controller</name>
```
Comment (JesusPoderoso) [inline test/unittest/dds/profiles/test_xml_profile.xml:305]: Update this name so the publish mode's flow controller name references an existing flow controller.

```suggestion
                        <name>test_default_flow_controller</name>
```

File: test/unittest/xmlparser/wrapper/XMLParserTest.hpp [hunk 2]
```diff
@@ -252,10 +252,18 @@ class XMLParserTest : public XMLParser
 
     static XMLP_ret getXMLThroughputController_wrapper(
             tinyxml2::XMLElement* elem,
-            ThroughputControllerDescriptor& throughputController,
+            ThroughputControllerDescriptor& throughtController,
```
Comment (JesusPoderoso) [inline test/unittest/xmlparser/wrapper/XMLParserTest.hpp:255]: I think this was not intended to be changed
```suggestion
            ThroughputControllerDescriptor& throughtputController,
```

File: test/unittest/xmlparser/wrapper/XMLParserTest.hpp [hunk 3]
```diff
@@ -252,10 +252,18 @@ class XMLParserTest : public XMLParser
...
             uint8_t ident)
     {
-        return getXMLThroughputController(elem, throughputController, ident);
+        return getXMLThroughputController(elem, throughtController, ident);
```
Comment (JesusPoderoso) [inline test/unittest/xmlparser/wrapper/XMLParserTest.hpp:258]: I think this was not intended to be changed
```suggestion
        return getXMLThroughputController(elem, throughtputController, ident);
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]

File: test/unittest/dds/profiles/test_xml_profile.xml (hunk 1 above)
Comment (Mario-DL) [reply test/unittest/dds/profiles/test_xml_profile.xml:305]: Good catch !

**Review by JesusPoderoso — APPROVED** [review]
LGTM with green CI

**Comment by JesusPoderoso** [comment]
@richiprosima please test this

**Comment by EduPonz** [comment]
@Mergifyio backport 2.13.x 2.10.x 2.6.x

File: resources/xsd/fastRTPS_profiles.xsd [hunk 4]
```diff
@@ -1831,6 +1835,41 @@
...
+        ├ scheduler             [flowControllerSchedulerPolicy],
+        ├ max_bytes_per_period  [int32],
+        ├ period_ms             [uint64],
+        └ sender_thread         [threadSettingsType]-->
+    <xs:complexType name="flowControllerDescriptorType">
+        <xs:all>
+            <xs:element name="name" type="string" minOccurs="1" maxOccurs="1"/>
+            <xs:element name="scheduler" type="flowControllerSchedulerPolicy" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="max_bytes_per_period" type="int32" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="period_ms" type="uint64" minOccurs="0" maxOccurs="1"/>
```
Comment (Ryanf55) [inline resources/xsd/fastRTPS_profiles.xsd:1857]: Some of the types merged in this PR do not exist. It broke the schema when it was backported to 2.6.x.
Any chance you can issue a patch? 

```
$ xmllint --schema fastRTPS_profiles.xsd --noout  myprofile.xml
fastRTPS_profiles.xsd:643: element element: Schemas parser error : element decl. '{http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles}max_bytes_per_period', attribute 'type': The QName value '{http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles}int32' does not resolve to a(n) type definition.
fastRTPS_profiles.xsd:644: element element: Schemas parser error : element decl. '{http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles}period_ms', attribute 'type': The QName value '{http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles}uint64' does not resolve to a(n) type definition.
fastRTPS_profiles.xsd:888: element element: Schemas parser error : element decl. '{http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles}tcp_negotiation_timeout', attribute 'type': The QName value '{http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles}uint32' does not resolve to a(n) type definition.
WXS schema fastRTPS_profiles.xsd failed to compile 
```

**Pull request merged**